components/binance/browser/binance_protocol_handler.cc

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/binance/browser/binance_protocol_handler.h"
+
+#include <string>
+#include <utility>
+
+#include "base/strings/strcat.h"
+#include "base/strings/string_util.h"
+#include "base/task/post_task.h"
+#include "brave/common/url_constants.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+#include "net/base/escape.h"
+
+namespace {
+
+GURL TranslateUrl(const GURL& url) {
+  if (!url.is_valid()) {
+    return GURL();
+  }
+
+  std::string path = url.path();
+  std::string query;
+
+  if (url.has_query()) {
+    query = base::StrCat({
```

Comment:
Just FYI, `net::AppendQueryParameter` is good for this kind of thing.

Reply:
ya `AppendQueryParameter ` it was used a lot of times in this PR but this file was copied from the rewards uphold protocol handler.